### PR TITLE
Remove duplicate padding on CupertinoFormSection

### LINKED ISSUE #128419: CupertinoFormSection header is wrapped by Padding twice, since Flutter 3.7.0
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Use `CupertinoFormSection`.
2. Use header parameter.
3. Compare Flutter v3.3.10 and Flutter v3.7.0

### Expected results

| | |
| --------------- | --------------- |
 | 

### Actual results

| | |
| --------------- | --------------- |
 | 

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(title),
      ),
      body: Column(
        children: [
          CupertinoFormSection(
            backgroundColor: Colors.transparent,
            header: const Text('This is section header'),
            children: const [
              ListTile(
                title: Text('This is ListTile title'),
              ),
            ],
          )
        ],
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.0, on macOS 13.4 22F66 darwin-arm64, locale en-JP)
[!] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[!] Android Studio (not installed)
[✓] IntelliJ IDEA Ultimate Edition (version 2023.1.2)
[✓] VS Code (version 1.77.3)
[✓] Connected device (4 available)
[✓] HTTP Host Availability

! Doctor found issues in 2 categories.

```

</details>

### PR DESCRIPTION
**Pull request by JaspervanRiet**
Fixes #128419, removes the duplicate padding that exists around the header and footer on `CupertinoFormSection`. This PR takes the implementation in #129065 and adds tests. I added the ones suggested in the PR review, as well as one more to test that `margin` is passed correctly from `CupertinoFormSection` to `CupertinoListSection`.

Note: I can't quite figure out if this is also a fix to #121543.

Potential review questions:

- Is the use of `offsetMoreOrLessEquals` here correct? I used it because of the vertical positioning. The horizontal positioning is always exact.

### Screenshots
<details>
  <summary>Before</summary>
  
image
</details>
<details>
  <summary>After</summary>
  
image
</details>

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/cupertino/form_section_test.dart [hunk 1]
```diff
@@ -168,4 +168,57 @@ void main() {
     final Iterable<RenderClipRRect> renderClips = tester.allRenderObjects.whereType<RenderClipRRect>();
     expect(renderClips, isEmpty);
   });
+
```
Comment (MitchellGoodwin) [inline packages/flutter/test/cupertino/form_section_test.dart:171]: Looks like the linter is complaining about a white space here.

File: packages/flutter/test/cupertino/list_section_test.dart [hunk 2]
```diff
@@ -223,4 +223,44 @@ void main() {
       }
     }
   });
+
```
Comment (MitchellGoodwin) [inline packages/flutter/test/cupertino/list_section_test.dart:226]: Linter says there's a white space here as well

**Review by MitchellGoodwin — APPROVED** [review]
LGTM with some easy linter issues. Thank you for putting these tests together.

For the record, this would not be a fix for #121543. I was originally wrong in thinking they were related, but the paddings in question are in separate places.

File: packages/flutter/test/cupertino/form_section_test.dart (hunk 1 above)
Comment (JaspervanRiet) [reply packages/flutter/test/cupertino/form_section_test.dart]: Cheers, fixed both and all good now 👍

**Comment by JaspervanRiet** [comment]
@MitchellGoodwin is this ready to merge?

**Review by Piinks — APPROVED** [review]
LGTM

**Pull request merged**